ui/src/common/components/AuthorAffiliation.jsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { Map } from 'immutable';
+import PropTypes from 'prop-types';
+import { INSTITUTIONS } from '../routes';
+import { getRecordIdFromRef, getInstitutionName } from '../utils';
+
+function AuthorAffiliation({ affiliation }) {
+  const institutionRecordId = getRecordIdFromRef(
+    affiliation.getIn(['record', '$ref'])
+  );
+
+  return institutionRecordId ? (
+    <Link to={`${INSTITUTIONS}/${institutionRecordId}`}>
+      {getInstitutionName(affiliation)}
```

Comment:
minor thing: `const institutionName = getInstitutionName(affiliation);` since it's used for both conditions.